test/grub2base_test.rb

```diff
@@ -348,4 +348,75 @@
       expect(subject.grub_default.kernel_params.serialize).to eq "console=ttyS1,4800n8"
     end
   end
+
+  describe "#merge" do
+    let(:other) do
+      other = Bootloader::Grub2Base.new
+      other.define_singleton_method(:name) { "grub2base" }
+      other
+    end
+
+    before do
+      subject.define_singleton_method(:name) { "grub2base" }
+    end
+
+    it "for default configuration prefer value from other if defined" do
+      subject.grub_default.default = "0"
+      other.grub_default.default = "saved"
+
+      subject.grub_default.terminal = :gfxterm
+
+      subject.grub_default.os_prober.enable
+      other.grub_default.os_prober.disable
+
+      subject.merge(other)
+
+      expect(subject.grub_default.default).to eq "saved"
+      expect(subject.grub_default.terminal).to eq :gfxterm
+      expect(subject.grub_default.os_prober).to be_disabled
+    end
+
+    it "for kernel line place subject params and then merged ones" do
```

Comment:
on kernel precedence is for later, that is why I use it this way. Do you have idea for better merge strategy? Problem is that rpm in post install script can add something to kernel command line and then we have trouble if we lost it.